arrays/arrays.c

```diff
@@ -34,9 +44,14 @@ Array *create_array (int capacity) {
  *****/
 void destroy_array(Array *arr) {
 
-  // Free all elements
+  if (arr->elements != NULL) { // free arr->elements if it exists
+    free(arr->elements); 
+  }
```

Comment:
you may need to to do a for loop to free all the elements. This only frees the array.